CONTRIBUTING.md

```diff
@@ -1,46 +1,56 @@
 
-Feel free to contribute by:
-- raising issues
-- participate in commenting/voting on the issues
-- create new PRs for Documentation or Code
+* If you are a first time contributor, please see the below [Steps to Contribute](#steps-to-contribute)
+* If you have documentation improvement ideas, go ahead and create a pull request. See [Pull Request checklist](#pull-request-checklist)
+* If you would like to make code contributions, please start with [Setting up the Development Environment](#setting-up-your-development-environment).
+* If you would like to work on something more involved, please connect with the OpenEBS Contributors. See [OpenEBS Community](https://github.com/openebs/openebs/tree/master/community)
 
-## Contributing to Source Code
+## Steps to Contribute
 
+OpenEBS is a Apache 2.0 Licensed project and all your commits should be signed with Developer Certificate of Origin. See [Sign your work](#sign-your-work). 
 
-### Setting up your Development Environment
+* Find an issue to work on or create a new issue. The issues are maintained at [openebs/openebs](https://github.com/openebs/openebs/issues). You can pick up from a list of [good-first-issues](https://github.com/openebs/openebs/labels/good%20first%20issue).
+* Claim your issue by commenting your intent to work on it to avoid duplication of efforts. 
+* Fork the repository on GitHub
+* Create a branch from where you want to base your work (usually master).
+* Make your changes. If you are working on code contributions, please see [Setting up the Development Environment](#setting-up-your-development-environment).
+* Relevant coding style guidelines are the [Go Code Review Comments](https://code.google.com/p/go-wiki/wiki/CodeReviewComments) and the _Formatting and style_ section of Peter Bourgon's [Go: Best Practices for Production Environments](http://peter.bourgon.org/go-in-production/#formatting-and-style).
+* Commit your changes by making sure the commit messages convey the need and notes about the commit.
+* Push your changes to the branch in your fork of the repository.
+* Submit a pull request to the original repository. See [Pull Request checklist](#pull-request-checklist)
 
-You will need an Linux host with Vagrant 1.9.1+ and Virtual Box 5.1. Follow these steps to setup your Development Environment.
 
-```
-cd <your-work-dir-on-linux-host>
-git clone https://github.com/openebs/maya.git
-vagrant up master-01
-vagrant ssh master-01
-#Only for the first time
-make bootstrap
-make deps
-make dev
-```
+## Pull Request Checklist
+* Rebase to the current master branch before submitting your pull request.
+* Commits should be as small as possible. Each commit should follow these checklist:
```

Comment:
follow these = follow the checklist below:

Reply:
done